lib/rubocop/cop/lint/to_json.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks to make sure `#to_json` includes an optional hash
+      # argument.  When overriding `#to_json`, callers may invoke JSON
+      # generation via `JSON.generate(your_obj)`.  Since `JSON#generate` allows
+      # for optional arguments, your method does too.
+      #
+      # @example
+      #   # bad
+      #   def to_json
+      #   end
+      #
+      #   # good
+      #   def to_json(opts = {})
+      #   end
+      #
+      class ToJSON < Cop
+        MSG = <<-M_JSON.freeze
+        `#to_json` requires an optional hash argument to be parsable via JSON.generate(obj).
+        M_JSON
+
+        def on_def(node)
+          return if node.method_name != :to_json
```

Comment:
Think these guard clauses would be clearer if combined:

```ruby
return unless node.method?(:to_json) && node.arguments.empty?
```

This reads exactly as the condition you laid out in the description. 🙂